test/test_DFT_driver.py

```diff
@@ -0,0 +1,369 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import pytest
+import time
+
+
+from maxwell.driver import *
+from maxwell.dg.mesh1d import *
+from maxwell.dg.dg1d import *
+from maxwell.fd.fd1d import *
+
+
+from nodepy import runge_kutta_method as rk
+
+def dft(x,freq,time):
+    X=[]
+    for f in range(len(freq)):
+        summatory=0.
+        for t in range(len(time)):
+            summatory=summatory + x[t] * np.exp(-2j*np.pi*freq[f]*time[t])
+        X.append(summatory)
+    return X
+
+# def test_DFT_driver():
+
+#     #Material distribution
+#     epsilon_1=6
+#     # epsilon_2=1
+#     L1=-50.0
+#     L2=50.0
+#     elements=100
+#     epsilons = epsilon_1*np.ones(elements)
+#     sigmas=np.zeros(elements) 
+#     sigmas[45:55]=1/8.
+#     # epsilons[int(elements/2):elements-1]=epsilon_2
+
+#     sp = DG1D(
+#         n_order=3,
+#         mesh=Mesh1D(L1, L2, elements, boundary_label="PEC"),
+#         epsilon=epsilons,
+#         sigma=sigmas
+#     )
+#     driver = MaxwellDriver(sp)
+
+#     #Type of wave
+#     s0 = 0.25
+#     x0=-2
+#     final_time = 1
+#     steps = 1000
+#     time_vector = np.linspace(0, final_time, steps)
+#     freq_vector = np.logspace(9, 10, 31)
+
+#     initialFieldE = np.exp(-(sp.x-x0)**2/(2*s0**2))
+#     initialFieldH = initialFieldE
+
+#     #Driver operates
+#     driver['E'][:] = initialFieldE[:]
+#     driver['H'][:] = initialFieldH[:]
+#     E_vector_1 = []
+#     E_vector_2=[]
+
+#     #DFT calculations
+#     for t in range(len(time_vector)):
+#         driver.run_until(time_vector[t])
+#         E_vector_1.append(driver['E'][2][0:45])
+#         E_vector_2.append(driver['E'][2][55:100])
+
+
+#     dft_E_1=dft(E_vector_1,freq_vector,time_vector)
+#     dft_E_2=dft(E_vector_2,freq_vector,time_vector)
+
+#     # Plot the DFTs
+#     plt.figure(figsize=(10, 6))
+#     plt.plot(freq_vector, np.abs(dft_E_1), label='', color='blue')
+#     plt.plot(freq_vector, np.abs(dft_E_2), label='', color='red')
+#     plt.xscale('log')  
+#     plt.xlabel('Frequency (Hz)')
+#     plt.ylabel('Magnitude')
+#     plt.title('DFT of Electric Fields')
+#     plt.legend()
+#     plt.grid(True)
+#     plt.show()
+
+@pytest.mark.skip(reason="Deactivated test, no analytical data to compare. Only checks visual behaviour.")
```

Comment:
Same as before. If this not really a test, this should not be its place. Consider moving it to a Jupyter notebook.